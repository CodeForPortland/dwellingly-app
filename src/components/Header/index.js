import React from "react";
import { Link, useLocation } from "react-router-dom";
import dwellinglylogo from "../../assets/images/dwellingly_logo_white.png";
import LogOutButton from "./LogOutButton";

const Header = (props) => {
  const loc = useLocation();

  if(loc.pathname === "/terms") {
    return null;
  }
	return (
    <header className="navbar bg-gradient">
      <Link className="navbar-item" id="header-logo" to="/">
        <img src={dwellinglylogo} alt="dwellingly logo" />
      </Link>
      <div className="logout-button">
        <LogOutButton />
      </div>
    </header>
  );
};

export default Header;
