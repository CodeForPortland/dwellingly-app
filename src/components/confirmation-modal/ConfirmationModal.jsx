import React from 'react';
import PropTypes from 'prop-types';
import Card from '../card/Card';
// import { CARD_TYPES } from '../../constants/constants';
import Icon from '../icon/Icon';
import './ConfirmationModal.scss';

class ConfirmationModal extends React.Component {
  render() {
    // Render nothing if the "show" prop is false
    if(!this.props.show) {
      return null;
    }

    return (
      <Card className="confirmCard" >
        <Card.Content className="card__content">
          <Icon icon="checkbox"/>
          <p className="title">
            {this.props.children}
          </p>
          <div className="button-container">
            <button className="cancel" onClick={this.props.onClose}
              type="button">
              Cancel
            </button>
            <button className="btn" onClick={this.props.onClose}
              type="button">
              Confirm
            </button>
          </div>
        </Card.Content>
      </Card>
    );
  }
}

ConfirmationModal.propTypes = {
  onClose: PropTypes.func.isRequired,
  show: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired
};

export default ConfirmationModal;
