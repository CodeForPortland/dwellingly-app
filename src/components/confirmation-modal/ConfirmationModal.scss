@import "../../constants/constants.scss";

.confirmCard {
  background: rgba(0, 0, 0, 0.6);
  justify-content: center;
  position: fixed;
  width: 100%;
  height: auto;

  .card__content {
    padding: 2.2em 2.5em;
    background: $color__white;
    width: 40%;
    border-radius: $border-radius--card;
    @include dialog_shadow;
    @include clearfix;
    align-self: center;
    text-align: center;

    .title {
      margin: 0;
      font-size: 1em;
    }
  }
}

.button-container {
  display: flex;
  justify-content: space-around;
  margin-top: 1.5em;

}

#yes-button {
  // margin-top: .5em;
  width: 8em;
}

.cancel {
  background: $color__gray--medium;
  border-radius: $border-radius--btn;
  color: $color__white;
  padding: 0 1rem;
  // margin-top: .5em;
  width: 8em;
}
