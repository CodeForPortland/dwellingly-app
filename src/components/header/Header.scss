header.appHeader{
  box-sizing: border-box;
  display: flex;
  justify-content: space-between;
  padding: 1rem;
  width: 100%;
}