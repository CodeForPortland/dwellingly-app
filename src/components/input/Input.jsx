import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import RadioArray from './RadioArray';
import Toggle from './Toggle';

import './Input.scss';

function InputContent({
  blockClass,
  children,
  id, // identifier for use with htmlFor and key properties
  label,
  maxLength, // maximum length
  model, // value of key in state
  name, // name of input
  onChange, // passes along the onChange function through props
  onClick, // for use with button/Link
  placeholder,
  type, // defines input type ["radio", "button", etc.]
  url, // for use with button/Link
  value // value from state, object, or string
}) {
  let inputContent = (
    <label htmlFor={id}>
      <span className={`${blockClass}__label`}>{label}</span>
      <input
        id={id}
        maxLength={maxLength || 200}
        name={name}
        onChange={onChange}
        placeholder={placeholder}
        type={type}
        value={value}
      />
    </label>
  );

  if (type === 'radio') {
    inputContent = (
      <label htmlFor={id}>
        <input
          checked={value === model}
          id={id}
          model={model}
          name={name}
          onChange={onChange}
          type="radio"
          value={value}
        />
        <span className="inline-input__radio-btn" />
        {label}
      </label>
    );
  }

  if (type === 'button') {
    inputContent = (
      <div>
        {!url && (
          <button aria-label={label} type="button" onClick={onClick}>
            {children}
          </button>
        )}
        {url && <Link to={url}>{children}</Link>}
      </div>
    );
  }

  if (type === 'date') {
    inputContent = (
      <label htmlFor={id}>
        <span className={`${blockClass}__label`}>{label}</span>
        <input
          id={id}
          name={name}
          onChange={onChange}
          type="datetime-local"
          value={value}
        />
      </label>
    );
  }

  return inputContent;
}

export default class Input extends Component {
  render() {
    const { blockClass, type, variants } = this.props;

    const variantsClasses =
      variants.length > 0
        ? variants
            .map(variant => `${blockClass}__${type}--${variant}`)
            .join(' ')
        : '';

    return (
      <div className={`input ${blockClass}`}>
        <div className={`${blockClass}__${type} ${variantsClasses}`}>
          <InputContent {...this.props} variantsClasses={variantsClasses} />
        </div>
      </div>
    );
  }
}

Input.propTypes = {
  blockClass: PropTypes.string,
  type: PropTypes.string,
  variants: PropTypes.arrayOf(PropTypes.string)
};

Input.defaultProps = {
  blockClass: 'inline-input',
  type: 'text',
  variants: []
};

export { Input, RadioArray, Toggle };
