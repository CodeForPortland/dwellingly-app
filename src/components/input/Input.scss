@import "../../constants/constants.scss";

/* F O R M S */
/* ==================================================== */
// INLINE-INPUT
.inline-input {
  // adding variable here as we use it for nearly all variants of .inline-input
  $inline-input__padding: 0.75rem 0;

  @include flex_align(space-between);
  border-bottom: 1px solid $color__gray--border;
  font-size: 1rem;
  margin: 0 1rem;
  position: relative;
  text-align: left;

  &:last-of-type {
    border: 0;
  }

  /* BUTTON */
  &__btn {
    text-align: center;
    width: 100%;

    a,
    button {
      display: block;
      font-size: 1em;
      padding: $inline-input__padding;
      text-align: center;
      width: 100%;
    }
  }

  /* RADIO BUTTON */
  &__radio {
    width: 100%;
    label {
      cursor: pointer;
      display: block;
      padding: $inline-input__padding;
      padding-left: 3em;
      position: relative;
    }

    input[type="radio"] {
      position: absolute;
      left: 0;
      opacity: 0;

      &:checked + .inline-input__radio-btn {
        background: $color__yellow;
      }
    }

    &-btn {
      border: 2px solid $color__yellow;
      border-radius: 50%;
      display: block;
      height: 1.25em;
      left: 0.75em;
      position: absolute;
      top: 0.825em;
      transition: background 0.2s ease;
      width: 1.25em;
    }
  }

  /* TEXT */
  &__tel,
  &__text,
  &__date,
  &__email,
  &__password {
    padding: 0.25em 0;
    width: 100%;

    &--full {
      label {
        @include flex_align(space-between);
      }

      .inline-input__label {
        display: inline-block;
        padding-right: 1em;
        width: 30%;
      }
      
      input {
        width: 70%;
      }
    }
  }

  /* DATE */
  &__date {
    padding: $inline-input__padding;

    input {
      font-weight: inherit;
    }
  }

  /* "TOGGLE" CHECKBOX */
  &__toggle {
    padding: 0.5rem 0;
    input[type="checkbox"],
    input[type="radio"] {
      display: none;

      &:checked + .inline-input__toggle__btn {
        background: $color__green;
        border-color: $color__green;

        &::after {
          left: 1.5em;
        }

        &:active {
          &::after {
            margin-left: -0.5em;
          }
        }
      }
    }

    &__btn {
      background: $color__white;
      border: 1px solid rgba(0, 0, 0, 0.1);
      border-radius: 2em;
      display: block;
      font-size: 1em;
      height: 2em;
      position: relative;
      text-indent: -9999em;
      transition: background 0.2s ease, border 0.2s ease;
      width: 3.5em;

      &::after {
        background: $color__white;
        border: 1px solid rgba(0, 0, 0, 0.1);
        border-radius: 1.5em;
        box-shadow: 0 3px 3px rgba(0, 0, 0, 0.05), 0 2px 2px rgba(0, 0, 0, 0.1),
          0 3px 1px rgba(0, 0, 0, 0.05);
        content: "";
        display: block;
        height: calc(2em - 2px);
        left: -1px;
        padding: 0;
        position: absolute;
        transition: all 0.2s cubic-bezier(0.175, 0.885, 0.32, 1.275);
        top: -1px;
        width: calc(2em - 2px);
        z-index: 1;
      }

      &:hover::after {
        will-change: padding;
      }

      &:active {
        &::after {
          padding-right: 0.5em;
        }
      }
    }
  }
}
/* end forms */
