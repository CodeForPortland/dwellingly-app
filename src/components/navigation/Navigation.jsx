import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from 'react-intl';
import UserControls from '../user-controls/UserControls';
import NavigationContent from './NavigationContent';
import Icon from '../icon/Icon';

import './Navigation.scss';
import { ROLES } from '../../constants/constants';

import { dummyUser } from '../../data';

class Navigation extends Component {
  constructor(props) {
    super(props);

    this.handleToggleMenu = this.handleToggleMenu.bind(this);

    this.state = {
      showMenu: false,
      user: dummyUser
    };
  }

  handleToggleMenu() {
    this.setState(({ showMenu }) => ({ showMenu: !showMenu }));
  }

  render() {
    const { desktopOnly, intl } = this.props;
    const { showMenu, user } = this.state;
    if (user.role !== ROLES.PROPERTY_MANAGER) {
      return (
        <div className="navigation">
          {desktopOnly && (
            <div className="frontmatter">
              <NavigationContent intl={intl} type="desktop" user={user} />
              <UserControls type="desktop" />
            </div>
          )}
          {!showMenu &&
            !desktopOnly && (
              <div className="actions">
                <button
                  type="button"
                  aria-label="Show Menu"
                  className="menu__btn action action--right"
                  onClick={this.handleToggleMenu}>
                  <Icon icon="menu" />
                </button>
              </div>
            )}

          {showMenu &&
            !desktopOnly && (
              <NavigationContent
                handleToggleMenu={this.handleToggleMenu}
                intl={intl}
                user={user}
              />
            )}
        </div>
      );
    }

    return null;
  }
}

Navigation.propTypes = {
  desktopOnly: PropTypes.bool,
  intl: intlShape.isRequired
};

Navigation.defaultProps = {
  desktopOnly: false
};

export default injectIntl(Navigation);
