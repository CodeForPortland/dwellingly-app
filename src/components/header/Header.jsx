import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './Header.scss';

class Header extends Component {
  static Label({ label, type, children }) {
    return (
      <div className={`hero__wrapper${type && ` hero__wrapper--${type}`}`}>
        {label && <h2>{label}</h2>}
        {children && children()}
      </div>
    );
  }

  render() {
    const { label, type, variant, variants, children } = this.props;
    const variantClasses =
      variants && variants.length > 0
        ? variants.map(variant => `hero--${variant}`).join(' ')
        : '';
    return (
      <header
        className={`hero ${
          variant ? `hero--${variant}` : ''
        } ${variantClasses}`}>
        {label && type && <Header.Label label={label} type={type} />}
        {children && children(...this.props)}
      </header>
    );
  }
}

Header.propTypes = {
  label: PropTypes.string,
  type: PropTypes.string,
  children: PropTypes.func,
  variant: PropTypes.string,
  variants: PropTypes.arrayOf(PropTypes.string)
};

Header.defaultProps = {
  children: undefined,
  label: undefined,
  type: undefined,
  variant: undefined,
  variants: undefined
};

export default Header;
