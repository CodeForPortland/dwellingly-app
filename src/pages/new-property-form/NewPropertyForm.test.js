import React from 'react';
import Input from '../../components/input/Input';
import Header from '../../components/header/Header';
import Search from '../../components/search/Search';
import ConfirmationModal from '../../components/confirmation-modal/ConfirmationModal';
import SuccessModal from '../../components/success-modal/SuccessModal';
import Navigation from '../../components/navigation/Navigation';

import renderer from 'react-test-renderer';
// import NewPropertyForm from './NewPropertyForm';

const ConfirmationModalTest = (props) =>
<ConfirmationModal
{...props}
/>;

const SuccessModalTest = (props) =>
<SuccessModal
{...props}
/>;

const SearchComponentTest = (props) =>
<Search
{...defaultProps}
{...props}
/>;

const NavigationComponentTest = (props) =>
<Navigation
{...defaultProps}
{...props}
/>;

describe('Input Component', () => {
  it('correctly renders Input component', () => {
    const InputComponent = renderer.create(<Input />).toJSON();
    expect(InputComponent).toMatchSnapshot();
  });
});

describe('Header Component', () => {
  it('correctly renders Header component', () => {
    const HeaderComponent = renderer.create(<Header />).toJSON();
    expect(HeaderComponent).toMatchSnapshot();
  });
});

describe('Search Component', () => {
  it('correctly renders Search component', () => {
    const SearchComponent = renderer.create(SearchComponentTest).toJSON();
    expect(SearchComponent).toMatchSnapshot();
  });
});

describe('ConfirmationModal Component', () => {
  it('correctly renders ConfirmationModal component', () => {
    const ConfirmationModalComponent = renderer.create(ConfirmationModalTest).toJSON();
    expect(ConfirmationModalComponent).toMatchSnapshot();
  });

  it('renders component when show prop is true', () => {
    const props = {
            show: true
        },
    ConfirmationModalComponent = mount(<ConfirmationModal {...props} />).find('.full-window-modal');
    expect(ConfirmationModalComponent.hasClass('full-window-modal')).toEqual(true);
  });
});

describe('SuccessModal Component', () => {
  it('correctly renders SuccessModal component', () => {
    const SuccessModalComponent = renderer.create(SuccessModalTest).toJSON();
    expect(SuccessModalComponent).toMatchSnapshot();
  });
});

describe('Navigation Component', () => {
  it('correctly renders Navigation component', () => {
    const NavigationComponent = renderer.create(NavigationComponentTest).toJSON();
    expect(NavigationComponent).toMatchSnapshot();
  });
});
