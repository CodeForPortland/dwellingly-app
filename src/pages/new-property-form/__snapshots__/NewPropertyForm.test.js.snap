// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal Component correctly renders ConfirmationModal component 1`] = `null`;

exports[`Header Component correctly renders Header component 1`] = `
<header
  className="hero  "
/>
`;

exports[`Input Component correctly renders Input component 1`] = `
<div
  className="input inline-input"
>
  <div
    className="inline-input__text "
  >
    <label>
      <span
        className="inline-input__label"
      />
      <input
        maxLength={200}
        type="text"
      />
    </label>
  </div>
</div>
`;

exports[`Navigation Component correctly renders Navigation component 1`] = `null`;

exports[`Search Component correctly renders Search component 1`] = `null`;

exports[`SuccessModal Component correctly renders SuccessModal component 1`] = `null`;
